specification/elasticsan/resource-manager/Microsoft.ElasticSan/stable/2023-01-01/elasticsan.json

```diff
@@ -1960,6 +2312,119 @@
+        },
+        "systemData": {
+          "description": "Resource metadata required by ARM RPC",
+          "readOnly": true,
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ]
+    },
+    "SnapshotCreateParameter": {
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "description": "Response for Volume Snapshot request.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of Volume Snapshot.",
+          "$ref": "#/definitions/SnapshotProperties"
+        },
+        "systemData": {
+          "description": "Resource metadata required by ARM RPC",
+          "readOnly": true,
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ]
+    },
+    "SnapshotList": {
+      "type": "object",
+      "description": "List of Snapshots",
+      "additionalProperties": false,
+      "properties": {
+        "value": {
+          "description": "An array of Snapshot objects.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Snapshot"
+          }
+        },
+        "nextLink": {
+          "description": "URI to fetch the next section of the paginated response.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "SnapshotProperties": {
+      "type": "object",
+      "required": [
+        "creationData"
+      ],
+      "description": "Properties for Snapshot.",
+      "properties": {
+        "creationData": {
+          "description": "Data used when creating a volume snapshot.",
+          "$ref": "#/definitions/SnapshotCreationData"
+        },
+        "provisioningState": {
+          "description": "State of the operation on the resource.",
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "sourceVolumeSizeGiB": {
+          "description": "Size of Source Volume",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        }
+      }
+    },
+    "SnapshotCreationData": {
+      "type": "object",
+      "required": [
+        "sourceUri"
+      ],
+      "description": "Data used when creating a volume snapshot.",
+      "properties": {
+        "sourceUri": {
+          "type": "string",
+          "description": "If createOption is Copy, this is the ARM id of the source snapshot or disk. If createOption is Restore, this is the ARM-like id of the source disk restore point."
+        }
+      }
+    },
+    "SnapshotUpdate": {
+      "type": "object",
+      "description": "Snapshot request.",
+      "properties": {
+        "tags": {
```

Comment:
>tags [](http://example.com/codeflow?start=9&length=4)

Snapshot seems to be a proxy type. Don't put tags on a proxy type: this is an ARM anti-pattern.

Reply:
Thanks @mentat9 .

Quick follow-up : how do we provide "tag-like" feature for proxy resources. In our case, we are expecting to have a large number of snapshots and hence we want to provide the user a way of classifying snapshots with appropriate tags.

Reply:
@mentat9 / @tadelesh can you please give us clarity on this too? Thanks

Reply:
@PreetPRG - It's OK to provide that kind of feature, but definitely don't call it "tags". Make sure it's obvious to clients that it's not the same as tags provided by ARM for tracked resources. It's common for APIs to provide a "metadata" or "category" property that clients can populate (or partially populate). You can find examples in other swaggers.